zephyr/CMakeLists.txt

```diff
@@ -43,52 +106,73 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
     --pad-header
     )
 
-  add_custom_command(
-    OUTPUT
-    ${signed_image_hex}
-    ${update_hex}
-    ${update_bin}
-    COMMAND
-    ${sign_cmd}
-    ${to_sign_hex}
-    ${signed_image_hex}
-    COMMAND
-    ${CMAKE_OBJCOPY}
-    --input-target=ihex
-    --output-target=binary
-    ${to_sign_hex}
-    ${to_sign_bin}
-    COMMAND
-    ${sign_cmd}
-    ${to_sign_bin}
-    ${update_bin}
-    COMMAND
-    ${sign_cmd}
-    --pad # This argument is needed for MCUboot to apply the test swap.
-    ${to_sign_hex}
-    ${test_update_hex}
-    COMMAND
-    ${sign_cmd}
-    ${to_sign_hex}
-    ${update_hex}
-    COMMAND
-    ${CMAKE_OBJCOPY}
-    --input-target=ihex
-    --output-target=ihex
-    --change-address $<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_SIZE>
-    ${update_hex}
-    ${PROJECT_BINARY_DIR}/moved_update.hex
-    DEPENDS
-    ${sign_depends}
+  set(app_offset $<TARGET_PROPERTY:partition_manager,PM_MCUBOOT_PRIMARY_SIZE>)
+
+  sign(${app_to_sign_hex}     # Hex to sign
+    ${PROJECT_BINARY_DIR}/app # Prefix for generated files
+    ${app_offset}             # Offset
+    ${app_sign_depends}       # Dependencies
+    app_signed_hex            # Generated hex output variable
     )
-  add_custom_target(mcuboot_sign_target DEPENDS ${signed_image_hex})
+
+  add_custom_target(mcuboot_sign_target DEPENDS ${app_signed_hex})
 
   set_property(GLOBAL PROPERTY
     mcuboot_primary_app_PM_HEX_FILE
-    ${signed_image_hex}
+    ${app_signed_hex}
     )
   set_property(GLOBAL PROPERTY
     mcuboot_primary_app_PM_TARGET
     mcuboot_sign_target
     )
+
+  if (CONFIG_MCUBOOT_BUILD_S1_VARIANT)
+    # Secure Boot (B0) is enabled, and we have to build update candidates
+    # for both S1 and S0.
+
+    foreach(slot s0_image;s1_image)
+      string(SUBSTRING ${slot} 0 2 slot_only) # 's0_image' -> 's0'
```

Comment:
How about `foreach(slot s0;s1)` instead? In that case, `${slot}` -> `${slot}_image` and `${slot_only}` -> `${slot}`.

Reply:
I see your point. But after making the change, I find it harder to see the relation between `${slot}` and `${slot}_image`. 

Earlier you could easily see if the variable being used was that of the full Sx slot, or just the image. I agree that the distinction is not clear to everyone, but that could be fixed using some docs.

Reply:
Ok, how about this?

    foreach(slot_only s0;s1)
        set(slot ${slot_only}_image)

Reply:
I agree

Reply:
Ok, good :). My suggestion:

    foreach(parent_slot s0;s1)
        set(slot ${parent_slot}_image)